src/index.test.ts

```diff
@@ -3788,176 +3791,2407 @@ describe('personalSign', function () {
   });
 });
 
-it('signTypedData and recoverTypedSignature V1 - single message', function () {
-  const address = '0x29c76e6ad8f28bb1004902578fb108c507be341b';
-  const privKeyHex =
-    '4af1bceebf7f3634ec3cff8a2c38e51178d5d4ce585c52d6043e5e2cc3418bb0';
-
-  const privKey = Buffer.from(privKeyHex, 'hex');
+describe('signTypedData', function () {
+  describe('V1', function () {
+    const signTypedDataV1Examples = {
+      // dynamic types supported by EIP-712:
+      bytes: [10, '10', '0x10', Buffer.from('10', 'utf8')],
+      string: [
+        'Hello!',
+        '0xbBbBBBBbbBBBbbbBbbBbbbbBBbBbbbbBbBbbBBbB',
+        '0xabcd',
+        '😁',
+      ],
+      // atomic types supported by EIP-712:
+      address: [
+        '0xbBbBBBBbbBBBbbbBbbBbbbbBBbBbbbbBbBbbBBbB',
+        '0xbBbBBBBbbBBBbbbBbbBbbbbBBbBbbbbBbBbbBBbBbBbBBBBbbBBBbbbBbbBbbbbBBbBbbbbBbBbbBBbB', // no apparent maximum length
+        '0x0',
+        10,
+        Number.MAX_SAFE_INTEGER,
+      ],
+      bool: [true, false, 'true', 'false', 0, 1, -1, Number.MAX_SAFE_INTEGER],
+      bytes1: [
+        '0x10',
+        10,
+        0,
+        1,
+        -1,
+        Number.MAX_SAFE_INTEGER,
+        Buffer.from('10', 'utf8'),
+      ],
+      bytes32: [
+        '0x10',
+        10,
+        0,
+        1,
+        -1,
+        Number.MAX_SAFE_INTEGER,
+        Buffer.from('10', 'utf8'),
+      ],
+      int8: [0, '0', '0x0', 255, -255],
+      int256: [0, '0', '0x0', Number.MAX_SAFE_INTEGER, Number.MIN_SAFE_INTEGER],
+      uint8: [0, '0', '0x0', 255, -255],
+      uint256: [
+        0,
+        '0',
+        '0x0',
+        Number.MAX_SAFE_INTEGER,
+        Number.MIN_SAFE_INTEGER,
+      ],
+      // atomic types not supported by EIP-712:
+      int: [0, '0', '0x0', Number.MAX_SAFE_INTEGER, Number.MIN_SAFE_INTEGER], // interpreted as `int256` by `ethereumjs-abi`
+      uint: [0, '0', '0x0', Number.MAX_SAFE_INTEGER, Number.MIN_SAFE_INTEGER], // interpreted as `uint256` by `ethereumjs-abi`
+      // `fixed` and `ufixed` types omitted because their encoding in `ethereumjs-abi` is very broken at the moment.
+      // `function` type omitted because it is not supported by `ethereumjs-abi`.
+    };
 
-  const typedData = [
-    {
-      type: 'string',
-      name: 'message',
-      value: 'Hi, Alice!',
-    },
-  ];
+    const signTypedDataV1ErrorExamples = {
+      string: [
+        {
+          input: 10,
```

Comment:
Second highlight - it doesn't accept numbers as string data. Makes sense, but it's unfortunate that V3 and V4 do.